### PR TITLE
Secure metrics server

### PR DESCRIPTION
**Pull request by spangenberg**

### CODE REVIEW

**Comment by enxebre** [comment]
@spangenberg  can we please make sure there's an e2e test that validates metrics are reported before merging this

**Comment by spangenberg** [comment]
@enxebre just validated positively that the `up{job="cluster-autoscaler-operator"}` metric is always available.

File: install/09_alertrules.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  groups:
+    - name: general.rules
+      rules:
+        - alert: ClusterAutoscalerOperatorDown
+          annotations:
+            message: ClusterAutoscalerOperator
+              discovery.
+          expr: |
+            absent(up{job="cluster-autoscaler-operator"} == 1)
+          for: 5m
```
Comment (enxebre) [inline install/09_alertrules.yaml]: why is this `for` value different everywhere
https://github.com/openshift/cluster-machine-approver/blob/master/manifests/0000_90_cluster-machine-approver_04_alertrules.yaml#L19

https://github.com/openshift/machine-api-operator/blob/master/install/0000_90_machine-api-operator_04_alertrules.yaml#L46

**Comment by enxebre** [comment]
/approve
other than https://github.com/openshift/cluster-autoscaler-operator/pull/119#discussion_r342557095 lgtm

Please hold cancel once the origin prometheus test are enable

**Pull request merged**